MetaMask/metamask-extension
# feat: Enable why did you render

## **Description**

1. What is the reason for the change?
- Enables the [why did you render](https://github.com/welldone-software/why-did-you-render) tool for the extension.
- this tool helps us more easily identify performance issues and unnecessary re renders
- this is a dev environment tool that should not be used in production.  
2. What is the improvement/solution?
- create a new ENV variable to enable/disable this tool. This variable is named `ENABLE_WHY_DID_YOU_RENDER`
- `this tool will disabled by default` (`-ENABLE_WHY_DID_YOU_RENDER: false`)
- when enabled, this tool profiles the entire app and logs for render reports in the browser console
- these are very useful when debugging performance issues as well as in day to day development when trying to reduce rerenders
- to install this tool I followed these setup instructions: https://github.com/welldone-software/why-did-you-render?tab=readme-ov-file#setup

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26339?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

#### Steps to ensure that this feature is disabled by default
1. yarn install && yarn start
9. load app in browser
10. cerate wallet
11. open browser console
12. There should be no render report logs in your console

#### Steps to test enabling the render reports
(assuming you already built the app in the previous steps)
1. open `builds.yml`
2. scroll to the bottom and find the `WHY_DID_YOU_RENDER_ENABLED` variable
4. set the value to `true`
5. run `yarn start`
6. reload the extension in the browser
7. open the browser console
8. you SHOULD see render reports for the component that you are on


## **Screenshots/Recordings**



### **Before**

N/A

### **After**
en example of a render report
<img width="681" alt="Screenshot 2024-08-13 at 11 20 41 AM" src="https://github.com/user-attachments/assets/49e6e1d9-ea12-419a-8bba-676fa36c2520">


video showing that this feature is `disabled` by default and can be turned on by changing the feature flag

https://github.com/user-attachments/assets/83c17ef9-94a7-44d7-b076-95de0d3a9b31

video showing general use case.

https://github.com/user-attachments/assets/9668db06-f193-4d1a-b5a3-ef7880514f40


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
